### ISSUE TITLE
Use Multi-GPU for training

### ISSUE THREAD

**Issue by miaoqiz** [jantic/DeOldify#238]
Hi,

Just wondering if the current code supports "DataParallel" features in Pytorch?

If "fastai" has similar support, what should I change in the code base?

Thanks!

**Comment by jantic** [comment]
If you want just simply DataParallel, Generator pretraining should be a simple conversion (haven't tried it lately, so be forewarned). For ColorizeTrainingArtistic that would be adding this code after getting the model

`model=torch.nn.DataParallel(model)`

In [here](https://github.com/jantic/DeOldify/blob/c6f28b9557cba1d38b580ae2bbfef2845103a7b3/deoldify/generators.py#L110)

But for the GAN part- that's just not easy and I haven't bothered trying to coerce it into working. [This ](https://towardsdatascience.com/how-to-train-a-gan-on-128-gpus-using-pytorch-9a5b27a52c73)looks like a good reference.

Now why haven't I bothered?  I think that's worth talking about.  First, if I'm going to do multi-gpu training, I do [distributed dataparallel training](https://docs.fast.ai/distributed.html) with SyncBatchNorm replacing the batchnorm layers. And I do this with my current work that isn't released to the public because I can get bigger batch sizes at bigger image sizes, but also training it in such a way that it's numerically the same as if they were on a single big gpu.  The key here in my mind is getting batch norm statistics combined- much better results that way.

But I can tell you this much- getting this to work for a generator is one thing, but a GAN is again more difficult. I won't be putting that work out as open source because it's a lot of extra work and only a few people will benefit from it.  Not to mention, it's a bit of a competitive advantage for us.
